File: ios/HybridVisionCameraPluginInatVisionFactory.swift
//
//  HybridVisionCameraPluginInatVisionFactory.swift
//
//  Created by Johannes Klein on 06.09.26.
//

import AVFoundation
import NitroModules

class HybridVisionCameraPluginInatVisionFactory: HybridVisionCameraPluginInatVisionFactorySpec {
  func createVisionCameraPluginInatVision(options: VisionCameraPluginInatVisionOptions) throws -> Promise<any HybridVisionCameraPluginInatVisionSpec> {
    return Promise.async {
      // Initialize our inference pipeline asynchronously
      return try HybridVisionCameraPluginInatVision(options: options)
    }
  }
}

File: android/src/main/java/com/margelo/nitro/visioncameraplugininatvision/HybridVisionCameraPluginInatVisionFactory.kt
package com.margelo.nitro.visioncameraplugininatvision

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip

@DoNotStrip
@Keep
class HybridVisionCameraPluginInatVisionFactory : HybridVisionCameraPluginInatVisionFactorySpec() {
  @DoNotStrip
  @Keep
  override fun createVisionCameraPluginInatVision(options: VisionCameraPluginInatVisionOptions): HybridVisionCameraPluginInatVisionSpec {
    return HybridVisionCameraPluginInatVision(options)
  }
}
